frontend/src/AppRoutes/index.tsx

```diff
@@ -51,179 +35,135 @@ import defaultRoutes, {
+				const orgName =
+					org && Array.isArray(org) && org.length > 0 ? org[0].name : '';
+
+				const { name, email, role } = user;
+
+				const identifyPayload = {
+					email,
+					name,
+					company_name: orgName,
+					role,
+					source: 'signoz-ui',
+				};
+
+				const sanitizedIdentifyPayload = pickBy(identifyPayload, identity);
+				const domain = extractDomain(email);
+				const hostNameParts = hostname.split('.');
+
+				const groupTraits = {
+					name: orgName,
+					tenant_id: hostNameParts[0],
+					data_region: hostNameParts[1],
+					tenant_url: hostname,
+					company_domain: domain,
+					source: 'signoz-ui',
+				};
+
+				window.analytics.identify(email, sanitizedIdentifyPayload);
+				window.analytics.group(domain, groupTraits);
+
+				posthog?.identify(email, {
+					email,
+					name,
+					orgName,
+					tenant_id: hostNameParts[0],
+					data_region: hostNameParts[1],
+					tenant_url: hostname,
+					company_domain: domain,
+					source: 'signoz-ui',
+					isPaidUser: !!licenses?.trialConvertedToSubscription,
+				});
+
+				posthog?.group('company', domain, {
+					name: orgName,
+					tenant_id: hostNameParts[0],
+					data_region: hostNameParts[1],
+					tenant_url: hostname,
+					company_domain: domain,
+					source: 'signoz-ui',
+					isPaidUser: !!licenses?.trialConvertedToSubscription,
+				});
+			}
+		},
+		[hostname, isFetchingLicenses, licenses, org],
+	);
 
+	// eslint-disable-next-line sonarjs/cognitive-complexity
 	useEffect(() => {
-		const isIdentifiedUser = getLocalStorageApi(LOCALSTORAGE.IS_IDENTIFIED_USER);
-
 		if (
-			isLoggedInState &&
+			!isFetchingLicenses &&
+			licenses &&
+			!isFetchingUser &&
 			user &&
-			user.userId &&
-			user.email &&
-			!isIdentifiedUser
+			!!user.email
 		) {
-			setLocalStorageApi(LOCALSTORAGE.IS_IDENTIFIED_USER, 'true');
-		}
-
-		if (
-			isOnBasicPlan ||
-			(isLoggedInState && role && role !== 'ADMIN') ||
-			!(isCloudUserVal || isEECloudUser())
-		) {
-			const newRoutes = routes.filter((route) => route?.path !== ROUTES.BILLING);
-			setRoutes(newRoutes);
-		}
+			const isOnBasicPlan =
+				licenses.licenses?.some(
+					(license) =>
+						license.isCurrent && license.planKey === LICENSE_PLAN_KEY.BASIC_PLAN,
+				) || licenses.licenses === null;
 
-		if (isCloudUserVal || isEECloudUser()) {
-			const newRoutes = [...routes, SUPPORT_ROUTE];
+			const isIdentifiedUser = getLocalStorageApi(LOCALSTORAGE.IS_IDENTIFIED_USER);
 
-			setRoutes(newRoutes);
-		} else {
-			const newRoutes = [...routes, LIST_LICENSES];
+			if (isLoggedInState && user && user.id && user.email && !isIdentifiedUser) {
+				setLocalStorageApi(LOCALSTORAGE.IS_IDENTIFIED_USER, 'true');
+			}
 
-			setRoutes(newRoutes);
+			let updatedRoutes = defaultRoutes;
+			// if the user is a cloud user
+			if (isCloudUserVal || isEECloudUser()) {
+				// if the user is on basic plan then remove billing
+				if (isOnBasicPlan) {
+					updatedRoutes = updatedRoutes.filter(
```

Comment:
@vikrantgupta25: Billing should be should only to admins, do we have a check somewhere else ?

Reply:
yes we have the same in `SideNav.tsx`